packages/alfa-rules/test/common/dom/get-colors.spec.tsx

```diff
@@ -289,6 +289,77 @@ test("getForeground() handles a mix of opacity and transparency and a linear gra
   });
 });
 
+test("getForeground() resolves `currentcolor` when color is set on parent", (t) => {
+  const target = <p>Content</p>;
+
+  h.document(
+    [
+      <html>
+        <div class="blue">
+          <div>
+            <div>
+              <div>
+                <div>
+                  <div>
+                    <div>
+                      <div>
+                        <div>Hello {target}!</div>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
```

Comment:
@Jym77 Is there a need to optimize for very deeply nested tree structures and if yes, has it been done before elsewhere?

Reply:
I think we're OK until we encounter some crazy cases 🙈 (plus, this is the same as `inherited` which is the default).